docs/conf.py

```diff
@@ -56,6 +56,7 @@ def setup(app):
     "site",
     "README.md",
     "release-notes/README.md",
+    "deploy/README.md",
```

Comment:
This has not been tested in the CI's `make -C docs upload-search-index` (which... I don't think that job should be run off every commit to `main` -- if you reviewers agree, then we should stop uploading to Algolia on success before we make the job succeed with this PR). Instead, I tested by hand with

```shell
python deploy/scrape.py --json
```

which instead of uploading the index to Algolia will just output it to stdout.

Reply:
The exclude patterns use fnmatch, so `deploy/*` is probably even better. Possibly in combination with */README.md

But to get things going quickly, this change is good.

Reply:
That said, uploading to Algolia seems like it'd be necessary to test search. I don't hate the idea of uploading only occasionally, but don't really have a strong opinion either way on that.

Reply:
> The exclude patterns use fnmatch, so deploy/* is probably even better. Possibly in combination with */README.md

Ooh -- I like it. You're right.

Reply:
> That said, uploading to Algolia seems like it'd be necessary to test search.

Aha! I possibly don't understand how that process works. If indexes are suitably versioned (which I hadn't considered at all), then running every commit seems safe enough to me. I'll investigate.